src/Lecture2.hs

```diff
@@ -70,7 +75,12 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt ind list = 
+    if ind >= (length list) || ind < 0
```

Comment:
Unfortunately, `length` hangs on infinite lists so this implementation of `removeAt` doesn't work on lists of infinite size ♾️